pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1221,3 +1221,16 @@ def test_aggregate_numeric_object_dtype():
         {"key": ["A", "B"], "col1": ["a", "c"], "col2": [0, 2]}
     ).set_index("key")
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_index_object_dtype():
+    # GH 40014
```

Comment:
can you add a 1-liner comment about the expected index here

Reply:
I wasnt quite not sure what you meant/expected, but I added a comment. Let me know if I misinterpreted something.